src/client/client.rs

```diff
@@ -546,26 +547,30 @@ impl Client {
 /// Start network connection to the server.
 async fn connect_stream(opts: &ClientOptions) -> Result<AsyncStream> {
     debug!("Connecting to {}:{}", opts.host, opts.port);
-    #[cfg(feature = "tls")]
-    match opts.tls_client_config {
-        Some(ref c) => {
+    match opts.connection_mode {
+        #[cfg(feature = "tls")]
+        ConnectionMode::Tls(ref c) => {
             let connector = TlsConnector::from(c.clone());
             let domain = DNSNameRef::try_from_ascii_str(&*opts.host)
                 .map_err(|e| Error::from_std_err(e))?;
             let tcp = TcpStream::connect((&*opts.host, opts.port)).await?;
             let conn = connector.connect(domain, tcp).await?;
             Ok(AsyncStream::TlsStream(conn))
         },
-        None => {
+        ConnectionMode::Tcp => {
             let tcp = TcpStream::connect((&*opts.host, opts.port)).await?;
             Ok(AsyncStream::TcpStream(tcp))
         }
-    }
-
-    #[cfg(not(feature = "tls"))]
-    {
-        let tcp = TcpStream::connect((&*opts.host, opts.port)).await?;
-        Ok(AsyncStream::TcpStream(tcp))
+        #[cfg(feature = "websocket")]
+        ConnectionMode::Websocket => {
+            // let url = format!("{}:{}", opts.host, opts.port);
+            // println!("Websocket: \"{}\"", url);
```

Comment:
Can we either remove these 2 lines, or if they're till useful log using `debug!`?

Reply:
Whoops. I thought I removed them, but I guess not.